src/mpm.ts

```diff
@@ -37,15 +38,21 @@ export async function setup(platform: string, architecture: string): Promise<str
         return Promise.reject(Error("Unable to find runner temporary directory."));
     }
     let mpmDest = path.join(runner_temp, `mpm${ext}`);
-    let mpm: string = await tc.downloadTool(mpmUrl, mpmDest);
+    if (!fs.existsSync(mpmDest)){
```

Comment:
If MPM already exists on the system, it might be better to delete it and redownload rather than skip downloading. I'm just thinking about self-hosted runners where they may have downloaded MPM a long time ago and its now a very out-of-date version.

Reply:
Makes sense. I updated the logic to delete the MPM file. Incidentally it also resolves the macos-14, R2023a job failure as well.